Modules/Loadable/Tables/Logic/vtkSlicerTablesLogic.cxx

```diff
@@ -68,27 +72,75 @@ ::AddTable(const char* fileName, const char* name)
     return 0;
     }
 
-  // Storage node
-  vtkNew<vtkMRMLTableStorageNode> tableStorageNode;
-  tableStorageNode->SetFileName(fileName);
-  this->GetMRMLScene()->AddNode(tableStorageNode.GetPointer());
-
   // Storable node
-  vtkNew<vtkMRMLTableNode> tableNode;
-  if (name)
-    {
-    tableNode->SetName(name);
-    }
-  this->GetMRMLScene()->AddNode(tableNode.GetPointer());
+  vtkMRMLTableNode *tableNode;
 
-  // Read
-  int res = tableStorageNode->ReadData(tableNode.GetPointer());
-  if (res == 0) // failed to read
+  if (std::string(fileName).find(".db") != std::string::npos)
```

Comment:
Determining file extension is not trivial, use the method that is added for this purpose: vtkMRMLStorageNode::GetLowercaseExtensionFromFileName

Example:
   // determine file type
  std::string extension = vtkMRMLStorageNode::GetLowercaseExtensionFromFileName(filename);
  if( extension.empty() )
    {
    vtkErrorMacro("ReadData: no file extension specified: " << filename.c_str());
    return false;
    }

  if ( !extension.compare(".nrrd")
      || !extension.compare(".nhdr")
      || !extension.compare(".mha")
      || !extension.compare(".mhd")
      || !extension.compare(".nii") )
    {
...